src/diffusers/loaders/textual_inversion.py

```diff
@@ -453,3 +453,91 @@ def load_textual_inversion(
             self.enable_sequential_cpu_offload()
 
         # / Unsafe Code >
+
+    def unload_textual_inversion(
+        self,
+        tokens: Optional[Union[str, List[str]]] = None,
+    ):
+        r"""
+        Unload Textual Inversion embeddings from the text encoder of [`StableDiffusionPipeline`]
+
+        Example:
+        ```py
+        from diffusers import AutoPipelineForText2Image
+        import torch
+
+        pipeline = AutoPipelineForText2Image.from_pretrained("runwayml/stable-diffusion-v1-5")
+
+        # Example 1
+        pipeline.load_textual_inversion("sd-concepts-library/gta5-artwork")
+        pipeline.load_textual_inversion("sd-concepts-library/moeb-style")
+
+        # Remove all token embeddings
+        pipeline.unload_textual_inversion()
+
+        # Example 2
+        pipeline.load_textual_inversion("sd-concepts-library/moeb-style")
+        pipeline.load_textual_inversion("sd-concepts-library/gta5-artwork")
+
+        # Remove just one token
+        pipeline.unload_textual_inversion("<moe-bius>")
+        ```
+        """
+
+        tokenizer = getattr(self, "tokenizer", None)
+        text_encoder = getattr(self, "text_encoder", None)
+
+        # Get textual inversion tokens and ids
+        token_ids = []
+        last_special_token_id = None
+
+        if tokens:
+            if isinstance(tokens, str):
+                tokens = [tokens]
+            for token_id, added_token in tokenizer.added_tokens_decoder.items():
+                if not added_token.special:
+                    if added_token.content in tokens:
+                        token_ids.append(token_id)
+                else:
+                    last_special_token_id = token_id
+            if len(token_ids) == 0:
+                raise ValueError("No tokens to remove found")
+        else:
+            tokens = []
+            for token_id, added_token in tokenizer.added_tokens_decoder.items():
+                if not added_token.special:
+                    token_ids.append(token_id)
+                    tokens.append(added_token.content)
+                else:
+                    last_special_token_id = token_id
+
+        # Delete from tokenizer
+        for token_id, token_to_remove in zip(token_ids, tokens):
+            del tokenizer._added_tokens_decoder[token_id]
+            del tokenizer._added_tokens_encoder[token_to_remove]
+        
+        # Fix token ids in tokenizer
+        key_id = 1
+        for token_id in tokenizer.added_tokens_decoder:
```

Comment:
can you explain why do we need this block?

Reply:
thanks for explaining this! @fabiorigano 

I'm not very familiar with the use case 
cc @apolinario ad @linoytsaban here can you take a look to see if we need to reorder the remaining added tokens after we remove some?

Reply:
Thanks @fabiorigano! I looked a bit but I'm actually not quite sure why it's necessary to reorder 🤔

Reply:
hi @linoytsaban, the reordering block is useful to have the same indeces as the text embeddings in the encoder, so multiple `unload_textual_inversion(<token-x>)` calls will remove the correct text embeddings. If the reordering is not done, when re-executing `unload_textual_inversion(<token-y>)` the last for loop may fail, because it may remove a different text embedding than expected.

Reply:
gotcha! that makes total sense, thanks for explaining! 🤗